docs/md/executor.md

```diff
@@ -0,0 +1,169 @@
+| `methodConfigurationVersion` | Expected version of the method configuration.                                                     |
+| `fromSource`                 | Instruction to coerce an output from an upstream `Source` to a type understood by this `executor`.|
+
+#### `workspace`
+A `{workspace-namespace}/{workspace-name}` string as it appears in the URL path
+in the Terra UI.
+
+Prerequisites:
+
+- The workspace must exist prior to workload creation.
+- `workflow-launcher@firecloud.org` must be a workspace "Owner" in order to
+  import snapshots to the workspace.
+- The workspace must be compatible with any downstream processing stage that
+  consumes its workflows.
+
+#### `methodConfiguration`
+A `{method-configuration-namespace}/{method-configuration-name}` string as it
+appears in the URL path in the Terra UI.
+
+Prerequisites:
+
+- The method configuration must exist within `workspace` prior to
+  workload creation.
+
+#### `methodConfigurationVersion`
+The expected version of `methodConfiguration`, stored as an integer
+in Firecloud.
+
+Prerequisites:
+
+- The `methodConfiguration` version when fetched from Firecloud should match
+  `methodConfigurationVersion`.
+  
+!!! warning "Implications of Version Mismatch"
+    A version mismatch may indicate a possible concurrent modification of the
+    method configuration used for launching submissions.  Modification is possible
+    programmatically or via the Terra UI.  An unexpected modification may cause
+    submission and/or workflow creation to fail.
+
+#### `fromSource`
+This attribute tells workflow-launcher how to coerce an output
+from an upstream `Source` into a data type understood by the executor.
+
+Prerequisites:
+
+- Must be one of the following supported coercion instructions.
+
+###### `importSnapshot`
+Workflow-launcher should import Terra Data Repository (TDR) snapshots
+into `workspace` as snapshot references,
+updating `methodConfiguration` with the reference as its root entity type.
+
+## Developer Guide
+An executor is a `Queue` that satisfies the `Executor` protocol below:
+```clojure
+(defprotocol Executor
+  (update-executor!
+    ^Executor
+    [^Queue    upstream  ;; The queue from which to pull items to execute
+     ^Executor executor  ;; This executor instance
+    ]
+    "Consume items from the `upstream` queue and enqueue
+     to the `executor` queue for consumption by a later processing stage,
+     performing any external effects as necessary.
+     Implementations should avoid maintaining in-memory state and making long-
+     running external calls, favouring internal queues to manage such tasks
+     asynchronously between invocations.  Note that the `Executor` and `Queue`
+     are parameterised types and the `Queue`'s parameterisation must be
+     convertible to the `Executor`s.")
+  (executor-workflows
+    ^WhatAmI?                 ;; TODO: type hint for workflow list
```

Comment:
Stuck on this, would love some help.

Reply:
FTR: `^PersistentVector` is about all one can do here.

Reply:
D'oh… I didn't push this before merging like I thought!

Will make a new PR to add.